File: api/schemas/project.py
# schemas/project.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime


class ProjectFileOut(BaseModel):
    id: int
    filename: str
    original_filename: str
    file_type: str
    file_size: int
    created_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class ProjectBase(BaseModel):
    name: str
    description: str | None = None
    code: str
    budget: float | None = None
    company_id: int | None = None
    project_type: str = "merkez"  # merkez | franchise
    address: str | None = None
    latitude: float | None = None
    longitude: float | None = None
    manager_name: str | None = None
    manager_phone: str | None = None
    manager_email: str | None = None
    is_active: bool = True


class ProjectCreate(ProjectBase):
    responsible_user_ids: list[int] = []


class ProjectUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    code: str | None = None
    budget: float | None = None
    company_id: int | None = None
    project_type: str | None = None
    address: str | None = None
    latitude: float | None = None
    longitude: float | None = None
    manager_name: str | None = None
    manager_phone: str | None = None
    manager_email: str | None = None
    is_active: bool | None = None
    responsible_user_ids: list[int] | None = None


class ProjectPersonnelOut(BaseModel):
    id: int
    full_name: str
    email: str
    role: str

    model_config = ConfigDict(from_attributes=True)


class ProjectOut(ProjectBase):
    id: int
    created_at: datetime | None = None
    updated_at: datetime | None = None
    project_files: list[ProjectFileOut] = []
    personnel: list[ProjectPersonnelOut] = []

    model_config = ConfigDict(from_attributes=True)

File: web/src/types/project.ts
// web/src/types/project.ts

export interface ProjectFile {
  id: number;
  filename: string;
  original_filename: string;
  file_type: string;
  file_size: number;
  created_at?: string;
}

export interface ProjectPersonnel {
  id: number;
  full_name: string;
  email: string;
  role: string;
}

export interface Project {
  id: number;
  name: string;
  description?: string;
  code: string;
  budget?: number;
  company_id?: number;
  project_type: "merkez" | "franchise";
  address?: string;
  latitude?: number;
  longitude?: number;
  manager_name?: string;
  manager_phone?: string;
  is_active: boolean;
  created_at?: string;
  updated_at?: string;
  project_files?: ProjectFile[];
  personnel?: ProjectPersonnel[];
}

export interface ProjectCreatePayload {
  name: string;
  description?: string;
  code: string;
  budget?: number;
  company_id?: number;
  project_type: "merkez" | "franchise";
  address?: string;
  latitude?: number;
  longitude?: number;
  manager_name?: string;
  manager_phone?: string;
  manager_email?: string;
  is_active?: boolean;
  responsible_user_ids?: number[];
}

export interface ProjectUpdatePayload {
  name?: string;
  description?: string;
  code?: string;
  budget?: number;
  company_id?: number;
  project_type?: "merkez" | "franchise";
  address?: string;
  latitude?: number;
  longitude?: number;
  manager_name?: string;
  manager_phone?: string;
  manager_email?: string;
  is_active?: boolean;
  responsible_user_ids?: number[];
}
